streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableReduceTest.java

```diff
@@ -44,9 +58,8 @@ public void shouldAddAndSubtract() {
                 this::differenceNotNullArgs
             ).get();
 
-
-        final InMemoryKeyValueStore<String, Set<String>> myStore =
-            new InMemoryKeyValueStore<>("myStore", null, null);
+        final InMemoryKeyValueStore underlyingStore = new InMemoryKeyValueStore("myStore");
+        final KeyValueStore<String, Set<String>> myStore = new KeyValueBytesStoreWrapper<>(underlyingStore, stringSerde, setSerde);
```

Comment:
Would it be possible to re-write the test not to need the set serde?

IIUC, the objective is just to make sure the reduce processor invokes add/subtract properly. Previously, it was convenient to use a `Set` to define this logic, but it's not so convenient anymore.

I don't have a problem with your implementation, it just seems like a huge chunk of code to do something we don't strictly need to do.

Alternatively, I just saw your `KeyValueStoreWrapper` below. You could make that store implementation available for all the tests that don't really need to test the bytes store logic. I think I'd prefer this approach.

Reply:
I'd agree, adding a `StringSetSerde` seems like an overkill here to me. I'd suggest we just use `String` as value, and reduce function like `v1+v2` (there are already some mock reducers in `o.a.k.test`) to separate the two values for easier validation.

Reply:
If we extend the KeyValueStoreWrapper idea and extract it for use by all the tests we basically end up with a copy of the old generic InMemoryKeyValueStore, that we use just for testing. Maybe that is actually the cleanest way to do things given how different tests were relying on it for different things..?

Reply:
Personally, I think it is cleaner, since tests like this one get to focus more on the logic under test (the processor) and less on managing the state store layers just right, which should be the responsibility of other tests.

Reply:
I also agree on the `StringSetSerde` is a little heavyweight.